cdqa/utils/converters.py

```diff
@@ -135,12 +157,54 @@ def pdf_converter(directory_path):
             df.loc[i] = [pdf, None]
             raw = parser.from_file(os.path.join(directory_path, pdf))
             s = raw["content"]
-            paragraphs = re.split(u"\n(?=\u2028|[A-Z-0-9])", s)
+            paragraphs = re.split("\n(?=\u2028|[A-Z-0-9])", s)
             list_par = []
+            temp_para = (
+                ""
+            )  # varible that stores paras with length<200 (considered as a line)
             for p in paragraphs:
-                if len(p) >= 200:
-                    list_par.append(p.replace("\n", ""))
-                df.loc[i, "paragraphs"] = list_par
+                if not p.isspace():  # checking if para is not only spaces
+                    if include_line_breaks:
+                        """
+                        If include_line_breaks is True, the condition
+                        will check on leght of paragraph.
+                        """
+                        if len(p) >= 200:
+                            """
+                            Before appending we check if variable
+                            temp_para is not empty which holds para with
+                            length<200 before current paragraph variable p
+                            """
+                            if temp_para:
+                                list_par.append(temp_para.strip())
+                                temp_para = (
+                                    ""
+                                )  # reset temp_para for new lines to be concatenated
+                                list_par.append(
+                                    p.replace("\n", "")
+                                )  # append current para with len>200
+                            else:
+                                list_par.append(p.replace("\n", ""))
+                        else:
+                            """
+                            else paragraph p (line) is appended to temp_para
+                            """
+                            line = p.replace("\n", " ").strip()
+                            temp_para = temp_para + f" {line}"
+                            continue
+                    else:
+                        """
+                        If include_line_breaks is set to False
+                        appending paragraph p as is to list_para
+                        """
+                        list_par.append(p.replace("\n", ""))
+                else:
+                    continue
+            else:
```

Comment:
What does this else do here? Do we really need it or if we remove nothing changes?

Reply:
Nothing changes. Removing it.